### DISCUSSION TITLE
Is it possible to search for the absence of an attribute?

### DISCUSSION THREAD

**Discussion by jcarres-mdsol** [grafana/tempo#2887] [Q&A] [answered]
Turns out we have some bad client implementations and we have server spans without `http.status_code` which makes them very useless. 
We want to hunt down every installation of this but I can't figure a query that would match them. Is it possible with the current TraceQl to do this?

**Comment by mapno** [comment, marked as answer]
Hi @jcarres-mdsol. Unfortunately, it's not possible yet. Currently only `{ .a != nil}` is supported.

A workaround to finding these spans could be using [span-metrics](https://grafana.com/docs/tempo/latest/metrics-generator/span_metrics/), adding `http.status_code` to its `dimensions` and doing the query with PromQL, which does support it `tempo_spanmetrics_total{http_status_code=""}`

**Comment by jcarres-mdsol** [comment]
Clever workaround but precisely we want to hunt down these implementations because we are planning to change one of the already available dimension `status` from `unset` to either `ok` or `error` and want to cover all the services.

I think we will add a processor to the opentelemetry collector that will check http.status_code and decide how to set status.code.
I think in practice this is how we will find these implementations, we will be using metrics to check what's still `unset` and then search for some example spans of those services and see why the processor was not able to set it to a value.

Thanks though, I did not know that `nil` was a special value which could be used.
